### DISCUSSION TITLE
forcing keydown/keyup/mousemove/mousedown events

### DISCUSSION THREAD

**Discussion by TheExecutionist0** [eaglerforge/EaglerForge#17] [Q&A]
I'm working on a mod that executes "keydown" events through the Gamepad API but the client itself is not picking up those forced events. Do I have to do document.dispatchEvent(...) or is there some sort of other variable I have to do this from?

**Comment by TheExecutionist0** [comment]
Just realized that dispatchEvent wont do shit because of isTrusted being set to false and read-only (stupid)

**Comment by stpv221** [comment]
murturtle did something like this in eaglertouch, IDK if it works. I know ur looking for specific keys, tho.
``` javascript
function(){ModAPI.mcinstance.$gameSettings.$keyBindForward.$pressed = 0}
```

**Reply by TheExecutionist0** [reply]
hm

**Reply by ZXMushroom63** [reply]
You can also avoid mcinstance by using the ModAPI.settings.keyBindings

**Reply by stpv221** [reply]
didn't know that was a thing, noted
